resolver/fqdn_only_resolver.go

```diff
@@ -0,0 +1,48 @@
+package resolver
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/0xERR0R/blocky/config"
+	"github.com/0xERR0R/blocky/model"
+	"github.com/0xERR0R/blocky/util"
+	"github.com/miekg/dns"
+)
+
+type FqdnOnlyResolver struct {
+	NextResolver
+	enabled bool
+}
+
+func NewFqdnOnlyResolver(cfg config.Config) ChainedResolver {
+	return &FqdnOnlyResolver{
+		enabled: cfg.FqdnOnly,
+	}
+}
+
+func (r *FqdnOnlyResolver) Resolve(request *model.Request) (*model.Response, error) {
+	if r.enabled {
+		domainFromQuestion := util.ExtractDomain(request.Req.Question[0])
+		if !strings.Contains(domainFromQuestion, ".") {
+			response := new(dns.Msg)
+			response.Rcode = dns.RcodeNameError
+
+			reason := fmt.Sprintf("NOTFQDN (%s)", domainFromQuestion)
```

Comment:
Reason is used as a label in prometheus metric (https://github.com/0xERR0R/blocky/blob/e313ef7fe868601b3b3757d41c3677954d0518b2/resolver/metrics_resolver.go#L48), this should be a limited amount of possible values (for example the group name for blocking or the address from upstream resolver). I think it is fine enough just to return "NOTFQDN" here